### PR TITLE
xpu support sharding stage3 and other minor fix

### PR DESCRIPTION
**Pull request by QingshuChen**
### PR types
Others

### PR changes
Others

### Description
xpu support sharding stage3 and other minor fix

### CODE REVIEW

File: python/paddle/distributed/fleet/meta_parallel/sharding/group_sharded_stage3.py [hunk 1]
```diff
@@ -89,8 +89,10 @@ def __init__(
         super().__init__()
 
         # Default configs
-        assert core.is_compiled_with_cuda() or (
-            device in core.get_all_custom_device_type()
+        assert (
+            core.is_compiled_with_cuda()
+            or core.is_compiled_with_xpu()
+            or (device in core.get_all_custom_device_type())
         ), "Only support CUDA / CustomDevice."
```
Comment (houj04) [inline python/paddle/distributed/fleet/meta_parallel/sharding/group_sharded_stage3.py]: 这里的`Only support CUDA / CustomDevice.`应该再加一个`XPU`。下个PR一起改吧。

**Review by houj04 — DISMISSED** [review]

File: cmake/external/xpu.cmake [hunk 2]
```diff
@@ -26,7 +26,7 @@ set(XPU_XPTI_LIB_NAME "libxpti.so")
 if(NOT DEFINED XPU_BASE_DATE)
   set(XPU_BASE_DATE "20230907")
 endif()
-set(XPU_XCCL_BASE_VERSION "1.0.53.6")
+set(XPU_XCCL_BASE_VERSION "1.0.62.3")
```
Comment (XiaociZhang) [inline cmake/external/xpu.cmake]: 62.3是zhiyuan分支出的，主线最新的是1.0.61.8

**Review by zhangyk0314 — APPROVED** [review]
LGTM

**Review by cqulilujia — APPROVED** [review]
LGTM

**Review by houj04 — APPROVED** [review]

**Pull request merged**